examples/switch_mode_persist.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/python3
+
+# Switch between modes without reallocating buffers.
+
+import time
+
+from picamera2 import Picamera2, Preview
+from picamera2.allocators import PersistentAllocator
+
+picam2 = Picamera2()
+picam2.allocator = PersistentAllocator()
```

Comment:
Super minor, would an `allocator` parameter to the constructor be tidy?

`picam2 = Picamera2(allocator=PersistentAllocator())`

Reply:
Yep, I'll add that